Makefile

```diff
@@ -176,14 +178,26 @@ else
 KUSTOMIZE=$(shell which kustomize)
 endif
 
+OPERATOR_SDK = $(BINDIR)/operator-sdk
+operator-sdk:
+ifeq (,$(wildcard $(OPERATOR_SDK)))
+	@{ \
+	set -e ;\
+	mkdir -p $(dir $(OPERATOR_SDK)) ;\
+	curl -sSLo $(OPERATOR_SDK) https://github.com/operator-framework/operator-sdk/releases/download/$(OPERATOR_SDK_VERSION)/operator-sdk_${OS}_${ARCH};\
+	chmod +x $(OPERATOR_SDK) ;\
+	}
+endif
+
 # Generate bundle manifests and metadata, then validate generated files.
 .PHONY: bundle
-bundle: manifests kustomize
-	operator-sdk generate kustomize manifests -q
+bundle: manifests kustomize operator-sdk
+	$(OPERATOR_SDK) generate kustomize manifests -q
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
-	$(KUSTOMIZE) build config/manifests | operator-sdk generate bundle -q --extra-service-accounts prometheus-k8s --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS) $(BUNDLE_FLAGS)
+	$(KUSTOMIZE) build config/manifests | $(OPERATOR_SDK) generate bundle -q --extra-service-accounts prometheus-k8s --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS) $(BUNDLE_FLAGS)
 	cp config/metadata/* $(OUTPUT_DIR)/metadata/
-	operator-sdk bundle validate $(OUTPUT_DIR)
+	$(OPERATOR_SDK) bundle validate $(OUTPUT_DIR)
+	sed -i "s/builder=.*/builder=operator-sdk-$(OPERATOR_SDK_VERSION)/" bundle.Dockerfile
```

Comment:
@leelavg Why do we need this? Up until now, some other mechanism was responsible to update the docker file annotations.

Reply:
some other mechanism --> I'm not sure what that mechanism does as here I'm only updating the bundle version to match sdk version as I was told there's an issue with the sdk version which updates builder version incorrectly.

- when we upgrade the sdk and no issues are observed i'll remove this line

``` diff
# gd bundle.Dockerfile 
diff --git a/bundle.Dockerfile b/bundle.Dockerfile
index bd3ac93..4789190 100644
--- a/bundle.Dockerfile
+++ b/bundle.Dockerfile
@@ -6,7 +6,7 @@ LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
 LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=ocs-osd-deployer
 LABEL operators.operatorframework.io.bundle.channels.v1=alpha
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.8.0
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.7.1+git
 LABEL operators.operatorframework.io.metrics.mediatype.v1=metrics+v1
 LABEL operators.operatorframework.io.metrics.project_layout=go.kubebuilder.io/v2
```

Reply:
You are fighting against the builder. If the builder wants `operator-sdk-v1.7.1+git` it should be `operator-sdk-v1.7.1+git` and not `operator-sdk-v1.8.0`

Reply:
ack, done.